### PR TITLE
Bug 2035704: RoleBindings list page filter doesn't apply

### PR DESCRIPTION
**Pull request by kdoberst**
Fixes [Bug 2035704](https://bugzilla.redhat.com/show_bug.cgi?id=2035704)

Fixes an issue where the filter was not applying when the filter was changed.  (First commit)

In addition, while working on this bug, I identified another bug where if a user selects "Cluster-wide RoleBindings" and then refreshes the page, this filter is not applied to the table.  The second commit fixes this issue.  The solution for this second issue may cause the filter bar to load slightly later than the data.

### CODE REVIEW

File: frontend/public/components/RBAC/bindings.jsx [hunk 1]
```diff
@@ -367,7 +367,7 @@ export const RoleBindingsPage = ({
       <ListPageBody>
         <ListPageFilter
           data={staticData}
-          loaded={loaded}
+          loaded={loaded && resources.ClusterRoleBinding.loaded}
```
Comment (spadgett) [inline frontend/public/components/RBAC/bindings.jsx:370]: Is there still potentially a problem if you select namespace bindings and reload the page? I'm wondering if we shouldn't simply change the meaning of `loaded` until everything is loaded even if it takes slightly longer for the table to appear.

Comment (kdoberst) [reply frontend/public/components/RBAC/bindings.jsx:370]: I agree we should show items once we know enough to show the correct filtering.  There is code to show the table if there is at least one item so I was wondering if performance was an issue.  Of course, showing incorrect information quickly isn't a great option either ;-)

Strangely enough, right now, items not showing on refresh is only an issue for the `Cluster-wide RoleBindings` option.  Here's why.  The items shown in the filter menu  list is dependent on a "hasCRBindings" variable: https://github.com/openshift/console/blob/master/frontend/public/components/RBAC/bindings.jsx#L336-L348.  The `Cluster-wide RoleBindings` is the only option that is different between the two filter menu lists.  

When the page loads and the items are initially filtered, the `hasCRBindings` variable is (or at least can be) false,  and the filter list without `Cluster-wide RoleBindings`  is loaded, therefore it is ignored.

I'll make the change so we wait until items are loaded before we show the table, just wanted to give future us an explanation if needed.

File: frontend/public/components/RBAC/bindings.jsx [hunk 2]
```diff
@@ -317,8 +317,9 @@ export const RoleBindingsPage = ({
 
   const data = React.useMemo(() => flatten(resources), [resources]);
 
-  // Show table if at least a RBAC loaded property is true
-  const loaded = Object.values(resources).some((r) => r.loaded);
+  // Show table if at least a RBAC loaded property is true and ClusterRoleBinding is loaded (this is needed to ensure filtering is correct)
+  const loaded =
+    Object.values(resources).some((r) => r.loaded) && resources.ClusterRoleBinding.loaded;
```
Comment (spadgett) [inline frontend/public/components/RBAC/bindings.jsx:322]: Won't this be the same as...?

```ts
  const loaded = resources.ClusterRoleBinding.loaded;
```

I'd probably just change it to

```suggestion
  const loaded = Object.values(resources).every((r) => r.loaded);
```

Comment (kdoberst) [reply frontend/public/components/RBAC/bindings.jsx]: 🤦‍♀️   Sorry - fixed now

File: frontend/public/components/RBAC/bindings.jsx [hunk 3]
```diff
@@ -317,8 +317,8 @@ export const RoleBindingsPage = ({
 
   const data = React.useMemo(() => flatten(resources), [resources]);
 
-  // Show table if at least a RBAC loaded property is true
-  const loaded = Object.values(resources).some((r) => r.loaded);
+  // Show table if at least a RBAC loaded property is true and ClusterRoleBinding is loaded (this is needed to ensure filtering is correct)
```
Comment (spadgett) [inline frontend/public/components/RBAC/bindings.jsx:320]: Looks like the comment is no longer correct?

**Review by jhadvig — APPROVED** [review]
Nice catch @kdoberst 👍 
/lgtm
/approve

**Pull request merged**